plugins/skip-button-plugin/src/skip-button.js

```diff
@@ -0,0 +1,74 @@
+import pillarbox from '@srgssr/pillarbox-web';
+
+/**
+ * @ignore
+ * @type {typeof import('video.js/dist/types/button').default}
+ */
+const Button = pillarbox.getComponent('Button');
+
+/**
+ * Represents a SkipButton component for the pillarbox player.
+ *
+ * @class SkipButton
+ */
+class SkipButton extends Button {
+  /**
+   * @type {TextTrackCue}
+   */
+  activeInterval;
+
+  constructor(player, options) {
+    super(player, options);
+    this.addClass('pbw-skip-button', 'vjs-hidden', 'vjs-visible-text');
+    this.player().on('srgssr/interval', ({ data }) => this.handleTimeIntervalChange(data));
+  }
+
+  handleClick(event) {
+    super.handleClick(event);
+    this.player().currentTime(this.activeInterval.endTime);
+  }
+
+  handleTimeIntervalChange(data) {
+    this.activeInterval = data;
+    if (!this.activeInterval) {
+      this.hide();
+
+      return;
+    }
+
+    /**
+     * @type {import('@srgssr/pillarbox-web/dist/types/src/dataProvider/model/typedef').TimeInterval}
+     */
+    const timeInterval = JSON.parse(data.text);
```

Comment:
I always trust the data too 🥲

Reply:
I've added some more typing documentation.